_observing-your-data/query-insights/top-n-queries.md

```diff
@@ -9,23 +9,28 @@ nav_order: 65
 
 Monitoring the top N queries in query insights features can help you gain real-time insights into the top queries with high latency within a certain time frame (for example, the last hour). 
 
-## Getting started
+## Configuring top N query monitoring
 
-To enable monitoring of the top N queries, configure the following [dynamic settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#dynamic-settings):
+You can configure monitoring top N queries by the following metric types:
 
-- `search.insights.top_queries.latency.enabled`: Set to `true` to [enable monitoring of the top N queries](#enabling-the-top-n-queries-feature).
-- `search.insights.top_queries.latency.window_size`: [Configure the window size](#configuring-window-size). 
-- `search.insights.top_queries.latency.top_n_size`: [Specify the value of n](#configuring-the-value-of-n).
+- `latency`
+- `cpu`
+- `memory`
 
-It's important to exercise caution when enabling this feature because it can consume system resources.
-{: .important}
+Each metric has a set of corresponding settings:
+
+- `search.insights.top_queries.<metric>.enabled`: Set to `true` to [enable top N query monitoring by the metric](#enabling-top-n-query-monitoring).
+- `search.insights.top_queries.<metric>.window_size`: [Configure the window size the top N queries by the metric](#configuring-the-window-size). 
+- `search.insights.top_queries.<metric>.top_n_size`: [Specify the value of N for the top N queries by the metric](#configuring-the-value-of-n).
 
+For example, to enable top N query monitoring by CPU usage, set `search.insights.top_queries.cpu.enabled` to `true`. For more information about ways to specify dynamic settings, see [Dynamic settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#dynamic-settings).
```

Comment:
"by CPU usage" => "of CPU usage"?

Reply:
No, you monitor queries that use the most CPU ("by")